event.go

```diff
@@ -534,3 +569,19 @@ func (er EventReference) MarshalJSON() ([]byte, error) {
 
 	return json.Marshal(&tuple)
 }
+
+// SplitID splits a matrix ID into a local part and a server name.
+func SplitID(sigil byte, id string) (local string, domain ServerName, err error) {
```

Comment:
Examples as tests please (they then show up as an example on godoc and are executed as tests, so win-win!).

They look like https://godoc.org/github.com/matrix-org/gomatrix#EncodeUserLocalpart and are implemented like https://github.com/matrix-org/gomatrix/blob/master/userids_examples_test.go#L5